akka-actor/src/main/scala/akka/io/InetAddressDnsResolver.scala

```diff
@@ -1,34 +1,68 @@
+/**
+ * Copyright (C) 2016-2017 Lightbend Inc. <http://www.lightbend.com>
+ */
 package akka.io
 
-import java.net.{ UnknownHostException, InetAddress }
+import java.net.{ InetAddress, UnknownHostException }
+import java.security.Security
 import java.util.concurrent.TimeUnit
 
 import akka.actor.Actor
 import com.typesafe.config.Config
 
 import scala.collection.immutable
-import sun.net.{ InetAddressCachePolicy ⇒ IACP }
 import akka.util.Helpers.Requiring
 
+import scala.util.Try
+
+/** Respects the settings that can be set on the Java runtime via parameters. */
 class InetAddressDnsResolver(cache: SimpleDnsCache, config: Config) extends Actor {
 
-  import IACP.NEVER // 0 constant
+  // Controls the cache policy for successful lookups only
+  private final val CachePolicyProp = "networkaddress.cache.ttl"
+  @deprecated("Deprecated property key, keeping for legacy compatibility; replaced by CachePolicyProp") 
+  private final val CachePolicyPropFallback = "sun.net.inetaddr.ttl"
+
+  // Controls the cache policy for negative lookups only
+  private final val NegativeCachePolicyProp = "networkaddress.cache.negative.ttl"
+  @deprecated("Deprecated property key, keeping for legacy compatibility; replaced by NegativeCachePolicyProp") 
+  private final val NegativeCachePolicyPropFallback = "sun.net.inetaddr.negative.ttl"
+
+  // default values (-1 and 0 are magic numbers, trust them)
+  private final val Forever = -1
+  private final val Never = 0
+  private final val DefaultPositive = 30
+
+  private lazy val cachePolicy: Int = {
+    val n = Try(Security.getProperty(CachePolicyProp).toInt)
+      .orElse(Try(Security.getProperty(CachePolicyPropFallback).toInt))
```

Comment:
This will silently fall back when the property is set to a value that is not an Int (such as `2s`)

Perhaps:
```
Option(Security.getProperty(CachePolicyProp))
  .orElse(Option(Security.getProperty(CachePolicyPropFallback))
  .map(_.toInt)
  .getOrElse(DefaultPositive)
```

Reply:
Tried to verify what Java does with invalid strings, but can't say if I managed to verify it silently ignores or if I just didn't use a path that touches the sys prop.